lib/node_modules/@stdlib/stats/mskmax/lib/main.js

```diff
@@ -0,0 +1,115 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var dtypes = require( '@stdlib/ndarray/dtypes' );
+var gmskmax = require( '@stdlib/stats/base/ndarray/mskmax' );
+var dmskmax = require( '@stdlib/stats/base/ndarray/dmskmax' );
+var smskmax = require( '@stdlib/stats/base/ndarray/smskmax' );
+var factory = require( '@stdlib/ndarray/base/binary-reduce-strided1d-dispatch-factory' );
+
+
+// VARIABLES //
+
+var idtypes = dtypes( 'real_and_generic' );
+var mdtypes = dtypes( 'integer' );
+var odtypes = dtypes( 'real_and_generic' );
+var policies = {
+	'output': 'promoted',
+	'casting': 'none'
+};
+var table = {
+	'types': [
+		'float64',
+		'uint8', // dmskmax: x, mask
+		'float32',
+		'uint8'  // smskmax: x, mask
+	],
+	'fcns': [
+		dmskmax,
+		smskmax
+	],
+	'default': gmskmax
+};
+
+
+// MAIN //
+
+/**
+* Computes the maximum value along one or more ndarray dimensions according to a mask.
+*
+* @name max
+* @type {Function}
+* @param {ndarray} x - input ndarray
+* @param {ndarray} mask - mask ndarray
+* @param {Options} [options] - function options
+* @param {IntegerArray} [options.dims] - list of dimensions over which to perform a reduction
+* @param {boolean} [options.keepdims=false] - boolean indicating whether the reduced dimensions should be included in the returned ndarray as singleton dimensions
+* @param {string} [options.dtype] - output ndarray data type
+* @throws {TypeError} first argument must be an ndarray-like object
+* @throws {TypeError} second argument must be an ndarray-like object
+* @throws {TypeError} options argument must be an object
+* @throws {RangeError} dimension indices must not exceed input ndarray bounds
+* @throws {RangeError} number of dimension indices must not exceed the number of input ndarray dimensions
+* @throws {Error} must provide valid options
+* @returns {ndarray} output ndarray
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+* var Uint8Array = require( '@stdlib/array/uint8' );
+* var ndarray = require( '@stdlib/ndarray/ctor' );
+*
+* // Create a data buffer:
+* var xbuf = new Float64Array( [ 1.0, 2.0, 3.0, 4.0, 5.0, 6.0 ] );
+*
+* // Create a mask buffer:
+* var mbuf = new Uint8Array( [ 0, 0, 0, 1, 0, 1 ] ); // eslint-disable-line id-length
```

Comment:
```suggestion
* var mbuf = new Uint8Array( [ 0, 0, 0, 1, 0, 1 ] );
```

Don't disable lint rules within JSDoc examples like this.

Reply:
Also, this lint rule is not even applicable here.